File: off_chain_data/application-typescript/src/getAllAssets.ts
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Client } from '@grpc/grpc-js';
import { connect } from '@hyperledger/fabric-gateway';
import { chaincodeName, channelName, newConnectOptions } from './connect';
import { AssetTransferBasic } from './contract';

export async function main(client: Client): Promise<void> {
    const connectOptions = await newConnectOptions(client);
    const gateway = connect(connectOptions);

    try {
        const network = gateway.getNetwork(channelName);
        const contract = network.getContract(chaincodeName);

        const smartContract = new AssetTransferBasic(contract);
        const assets = await smartContract.getAllAssets();
        const assetsJson = JSON.stringify(assets, undefined, 2);
        assetsJson.split('\n').forEach(line => console.log(line)); // Write line-by-line to avoid truncation
    } finally {
        gateway.close();
    }
}

File: off_chain_data/application-java/app/src/main/java/GetAllAssets.java
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import io.grpc.Channel;
import org.hyperledger.fabric.client.CommitException;
import org.hyperledger.fabric.client.CommitStatusException;
import org.hyperledger.fabric.client.EndorseException;
import org.hyperledger.fabric.client.Gateway;
import org.hyperledger.fabric.client.SubmitException;

import java.io.IOException;
import java.security.InvalidKeyException;
import java.security.cert.CertificateException;

public final class GetAllAssets implements Command {
    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();

    @Override
    public void run(final Channel grpcChannel)
            throws CertificateException, IOException, InvalidKeyException, EndorseException, CommitException, SubmitException, CommitStatusException {
        try (Gateway gateway = Connections.newGatewayBuilder(grpcChannel).connect()) {
            var network = gateway.getNetwork(Connections.CHANNEL_NAME);
            var contract = network.getContract(Connections.CHAINCODE_NAME);

            var smartContract = new AssetTransferBasic(contract);

            var assets = smartContract.getAllAssets();
            var assetsJson = GSON.toJson(assets);
            System.out.println(assetsJson);
        }
    }
}
